WalletWasabi/Services/Terminate/TerminateService.cs

```diff
@@ -20,6 +22,21 @@ public TerminateService(Func<Task> terminateApplicationAsync)
 			TerminateApplicationAsync = terminateApplicationAsync;
 			AppDomain.CurrentDomain.ProcessExit += CurrentDomain_ProcessExit;
 			Console.CancelKeyPress += Console_CancelKeyPress;
+
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				SystemEvents.SessionEnding += Windows_SystemEvents_SessionEnding;
+			}
+		}
+
+		private void Windows_SystemEvents_SessionEnding(object sender, SessionEndingEventArgs e)
```

Comment:
```suggestion
		/// <summary>
		/// Occurs when the user is trying to log off or shut down the system.
		/// </summary>
		private void Windows_SystemEvents_SessionEnding(object sender, SessionEndingEventArgs e)
```

as per https://docs.microsoft.com/en-us/dotnet/api/microsoft.win32.systemevents.sessionending?view=dotnet-plat-ext-3.1

Reply:
What does this function do really? How will it behave for a user?

Reply:
This event is triggered when the mentioned events happen. So it will start the disposal. After ~7 seconds the OS kills the process event if e.Cancel is true.

Reply:
This should a remark in the documentation 👍 it's very useful to know.